unit-tests/src/test/java/org/eclipse/collections/impl/collector/Collectors2Test.java

```diff
@@ -1111,6 +1112,34 @@ public void countByParallel()
         Assert.assertEquals(50000, counts.occurrencesOf(1));
     }
 
+    @Test
+    public void countByEach()
+    {
+        List<Interval> intervals = FastList.newListWith(
+                Interval.evensFromTo(1, 100),
+                Interval.oddsFromTo(1, 100));
+
+        MutableBag<Integer> counts = intervals.stream().collect(Collectors2.countByEach(iv -> iv.collect(i -> i % 2).toList()));
+
+        Assert.assertEquals(Interval.oneTo(100).countBy(i -> i % 2), counts);
+        Assert.assertEquals(50, counts.occurrencesOf(0));
+        Assert.assertEquals(50, counts.occurrencesOf(1));
+    }
+
+    @Test
+    public void countByEachParallel()
+    {
+        List<Interval> intervals = FastList.newListWith(
+                Interval.evensFromTo(1, 100000),
+                Interval.oddsFromTo(1, 100000));
+
+        MutableBag<Integer> counts = intervals.stream().parallel().collect(Collectors2.countByEach(iv -> iv.collect(i -> i % 2).toList()));
```

Comment:
Same comment on toList() being unnecessary.

Reply:
Sure, let me include this in the final squashed commit